modules/hal_nordic/Kconfig

```diff
@@ -204,6 +204,7 @@ config NRF_802154_SECURITY_KEY_STORAGE_SIZE
 
 config NRF_802154_CARRIER_FUNCTIONS
 	bool "nRF 802.15.4 carrier functions"
+	default y if OPENTHREAD_DIAG
```

Comment:
Should `OPENTHREAD_DIAG` `depend on NRF_802154_CARRIER_FUNCTIONS`? Someone could disable this default which, if I understand correctly, would break the build